.pre-commit-config.yaml

```
@@ -19,6 +19,16 @@ repos:
       - id: trailing-whitespace
         types_or: [c, python, rst]
 
+  - repo: local
+    hooks:
+      - id: c-file-whitespace
+        name: "Check C file whitespace"
+        entry: "python Tools/patchcheck/untabify.py"
+        language: "system"
+        types_or: ['c', 'c++']
```

We were running on `.c` and `.h` files before:
```python
c_files = [fn for fn in file_paths if fn.endswith(('.c', '.h'))]
```
Both of those are matched by the `c` type, so this would be closer to parity:

```suggestion
        types_or: [c]
```

We do have half a dozen `.cpp` files in the codebase, do we want to expand to include them? Should we also add `c++` for `trailing-whitespace` above?

---

The test passed with 'c++' included, I imagined that it was a previous oversight that they weren't included. I'll check if the trailing whitespace check also passes.

---

``Tools/msi/bundle/bootstrap/PythonBootstrapperApplication.cpp`` fails with 16 lines changed. Other than that all good (@zooba do you want to enable the trailing whitespace check here? No real views either way, if you'd prefer to keep the whitespace then that's the status quo anyway!)

A

---

(We can always remove `c++` later if needed.)